### PR TITLE
Add auto merge github action

### PR DESCRIPTION
**Pull request by ralren**
Created a GH action that will auto merge the PR after approval.

### CODE REVIEW

File: .github/workflows/auto-approve.yml [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
+          go mod tidy
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+      - name: set up Git
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+        if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
```
Comment (chrisgilmerproj) [inline .github/workflows/auto-approve.yml:41]: Can you put this `if` statement at the top? I feel like I remember Github fixing it so you could. Meaning you can have:

```
  tidy:
    name: run go mod tidy and updated
    runs-on: ubuntu-latest
    if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
...
```

If that works you can reduce a ton copied code.

Comment (ralren) [reply .github/workflows/auto-approve.yml:41]: ooh will try

File: .github/workflows/auto-approve.yml [hunk 2]
```diff
@@ -14,14 +14,14 @@ jobs:
...
     runs-on: ubuntu-latest
+    if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
     steps:
       - name: checkout
         uses: actions/checkout@v1
         if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
       - name: reattach HEAD to Head Ref
         # b/c checkout action leaves in detached head state https://github.com/actions/checkout/issues/6
         run: git checkout "$(echo ${{ github.head_ref }})"
-        if: github.head_ref != '' && (github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]')
```
Comment (chrisgilmerproj) [inline .github/workflows/auto-approve.yml]: You may still want to retain `if: github.head_ref != ''` on this line.

Comment (ralren) [reply .github/workflows/auto-approve.yml]: darn it

Comment (ralren) [reply .github/workflows/auto-approve.yml]: ok. getting rid of it of the previous if and adding this one back in.

File: .github/workflows/auto-approve.yml [hunk 3]
```diff
@@ -1,13 +1,72 @@
...
+    name: run go mod tidy and updated
+    runs-on: ubuntu-latest
+    if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1
+      - name: reattach HEAD to Head Ref
+        # b/c checkout action leaves in detached head state https://github.com/actions/checkout/issues/6
+        run: git checkout "$(echo ${{ github.head_ref }})"
+        if: github.head_ref != '' && (github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]')
```
Comment (chrisgilmerproj) [inline .github/workflows/auto-approve.yml:24]:
```suggestion
        if: github.head_ref != ''
```

I think you only need the one piece. I am hoping there's a follow on effect from the above `if` statement that you moved on line 17.

Comment (ralren) [reply .github/workflows/auto-approve.yml:24]: let's find out!

**Pull request merged**